pkg/controllers/machine/lifecycle/launch.go

```diff
@@ -65,13 +64,17 @@ func (l *Launch) Reconcile(ctx context.Context, machine *v1alpha5.Machine) (reco
 	if err != nil || created == nil {
 		return reconcile.Result{}, err
 	}
-	l.cache.SetDefault(client.ObjectKeyFromObject(machine).String(), created)
+	l.cache.SetDefault(string(machine.UID), created)
```

Comment:
Why do we need to type cast this here? Should we either have UID be a string or we should have a String() method?

Reply:
UID is an upstream concept. We have to type-cast. There's not good way around this